File: cpp-functions/calculator-using-functions.py
def addition(num1, num2):
    return num1 + num2


def subtraction(num1, num2):
    return num1 - num2


def multiplication(num1, num2):
    return num1 * num2


def division(num1, num2):
    return num1 / num2


def floor_division(num1, num2):
    return num1 // num2


def modulus(num1, num2):
    return num1 % num2


def exponent(num1, num2):
    return num1**num2


print("\n-- Welcome to calculator --")

while True:

    print("-" * 40)
    first_number = int(input("Enter the first number: "))
    second_number = int(input("Enter the second number: "))
    print("Press 1 for addition")
    print("Press 2 for subtraction")
    print("Press 3 for multiplication")
    print("Press 4 for division")
    print("Press 5 for floor division")
    print("Press 6 for modulus")
    print("Press 7 for exponent")
    print("Press 0 for quit")
    print("-" * 40)

    user_choice = int(input("Enter your choice: "))
    if user_choice == 1:
        result = addition(first_number, second_number)
        print(f"Sum of {first_number} + {second_number} = {result}")
    elif user_choice == 2:
        result = subtraction(first_number, second_number)
        print(f"Subtraction {first_number} - {second_number} = {result}")
    elif user_choice == 3:
        result = multiplication(first_number, second_number)
        print(f"Multiplication {first_number} x {second_number} = {result}")
    elif user_choice == 4:
        if second_number == 0:
            print("0 is not allowed for division, try again!")
        else:
            result = division(first_number, second_number)
            print(f"Division {first_number} / {second_number} = {result}")
    elif user_choice == 5:
        if second_number == 0:
            print("0 is not allowed for floor division, try again!")
        else:
            result = floor_division(first_number, second_number)
            print(f"Floor Division {first_number} // {second_number} = {result}")
    elif user_choice == 6:
        result = modulus(first_number, second_number)
        print(f"Modulus {first_number} % {second_number} = {result}")
    elif user_choice == 7:
        result = exponent(first_number, second_number)
        print(f"Exponent {first_number} ** {second_number} = {result}")
    elif user_choice == 0:
        print("You have exited the calculator, Bye!")
        break
    else:
        print("Invalid input, try again!")

File: cpp-functions/calculator-using-functions.cpp
#include <iostream>
#include <math.h>

using namespace std;

int addition(int num1, int num2)
{
  return num1 + num2;
}

int subtraction(int num1, int num2)
{
  return num1 - num2;
}

int multiplication(int num1, int num2)
{
  return num1 * num2;
}

float division(int num1, int num2)
{
  return static_cast<float>(num1) / num2;
}

int floor_division(int num1, int num2)
{
  return num1 / num2;
}

int exponent(int num1, int num2)
{
  return pow(num1, num2);
}

int main()
{
  cout << "\n-- Welcome to calculator --" << endl;
  while (true)
  {
    int a, b, user_choice;
    float result;

    cout << "----------------------------------------" << endl;
    cout << "Enter the first number: ";
    cin >> a;
    cout << "Enter the second number: ";
    cin >> b;
    cout << "Press 1 for addition" << endl;
    cout << "Press 2 for subtraction" << endl;
    cout << "Press 3 for multiplication" << endl;
    cout << "Press 4 for division" << endl;
    cout << "Press 5 for floor division" << endl;
    cout << "Press 6 for modulus" << endl;
    cout << "Press 7 for exponent" << endl;
    cout << "Press 0 to quit" << endl;
    cout << "----------------------------------------" << endl;
    cout << "Enter your choice: ";
    cin >> user_choice;

    if (user_choice == 1)
    {
      result = addition(a, b);
      cout << "Sum of " << a << " + " << b << " = " << result << endl;
    }
    else if (user_choice == 2)
    {
      result = subtraction(a, b);
      cout << "Subtraction of " << a << " - " << b << " = " << result << endl;
    }
    else if (user_choice == 3)
    {
      result = multiplication(a, b);
      cout << "Multiplication of " << a << " x " << b << " = " << result << endl;
    }
    else if (user_choice == 4)
    {
      if (b == 0)
      {
        cout << "Error! Division by zero is not allowed." << endl;
      }
      else
      {
        result = division(a, b);
        cout << "Division of " << a << " / " << b << " = " << result << endl;
      }
    }
    else if (user_choice == 5)
    {
      if (b == 0)
      {
        cout << "Error! Division by zero is not allowed." << endl;
      }
      else
      {
        result = floor_division(a, b);
        cout << "Floor Division of " << a << " // " << b << " = " << result << endl;
      }
    }
    else if (user_choice == 6)
    {
      result = a % b;
      cout << "Modulus of " << a << " % " << b << " = " << result << endl;
    }
    else if (user_choice == 7)
    {
      result = exponent(a, b);
      cout << "Exponent of " << a << " ** " << b << " = " << result << endl;
    }
    else if (user_choice == 0)
    {
      cout << "You have exited the calculator. Bye!" << endl;
      break;
    }
    else
    {
      cout << "Invalid input, please try again!" << endl;
    }
  }
  return 0;
}
